src/Joystick/Joystick.cc

```diff
@@ -30,11 +30,14 @@ const char* Joystick::_rgFunctionSettingsKey[Joystick::maxFunction] = {
     "ThrottleAxis"
 };
 
-Joystick::Joystick(const QString& name, int axisCount, int buttonCount, MultiVehicleManager* multiVehicleManager)
+Joystick::Joystick(const QString& name, int axisCount, int buttonCount, int hatCount, MultiVehicleManager* multiVehicleManager)
     : _exitThread(false)
     , _name(name)
     , _axisCount(axisCount)
     , _buttonCount(buttonCount)
+    , _hatCount(hatCount)
+    , _hatButtonCount(4*hatCount)
```

Comment:
Where does this come from? Seems a bit magic. Hats can have buttons, but the actual number is not reported?